needle-slack-connector/src/app/_components/providers/SlackResourcesProvider.tsx

```diff
@@ -0,0 +1,144 @@
+  workspace: SlackWorkspace | undefined;
+  userTimezone: TimezoneInfo | undefined;
+  isLoading: boolean;
+};
+
+const SlackResourcesContext = createContext<SlackResourcesContextType | null>(
+  null,
+);
+
+interface SlackResourcesProviderProps {
+  children: React.ReactNode;
+  credentials: string;
+  userId: string;
+}
+
+export function SlackResourcesProvider({
+  children,
+  credentials,
+  userId,
+}: SlackResourcesProviderProps) {
+  const [selectedChannels, setSelectedChannels] = useState<SlackChannel[]>([]);
+
+  const { data: workspace, isLoading: isWorkspaceLoading } =
+    api.connectors.getWorkspaces.useQuery({
+      accessToken: credentials,
+    });
+
+  const { data: channels, isLoading: isChannelsLoading } =
+    api.connectors.getChannels.useQuery({
+      accessToken: credentials,
+    });
+
+  const { data: userTimezone, isLoading: isTimezoneLoading } =
+    api.connectors.getUserTimezone.useQuery({
+      accessToken: credentials,
+      userId,
+    }) as { data: TimezoneInfo | undefined; isLoading: boolean };
+
+  const { data: messages, isLoading: isMessagesLoading } =
+    api.connectors.getMessages.useQuery(
+      selectedChannels.length > 0
+        ? {
+            accessToken: credentials,
+            channelIds: selectedChannels.map((channel) => channel.id),
+          }
+        : skipToken,
+      {
+        enabled: selectedChannels.length > 0,
+      },
+    );
+
+  const mappedMessages = messages?.flatMap(
+    (channelMessages, index) =>
+      channelMessages.messages.map((msg) => ({
+        ...msg,
+        channelName: selectedChannels[index]?.name,
+        channelId: selectedChannels[index]?.id,
+      })) as SlackMessage[],
+  );
+
+  const { data: canvases } = api.connectors.getCanvases.useQuery(
+    selectedChannels.length > 0
+      ? {
+          accessToken: credentials,
+          channelId: selectedChannels[selectedChannels.length - 1]?.id ?? "",
+        }
+      : skipToken,
+  );
+
+  useEffect(() => {
+    if (canvases && selectedChannels.length > 0) {
+      const lastSelectedChannel = selectedChannels[selectedChannels.length - 1];
+
+      if (canvases && selectedChannels.length > 0 && lastSelectedChannel) {
+        const updatedChannel: SlackChannel = {
+          id: lastSelectedChannel.id,
+          name: lastSelectedChannel.name,
+          canvases:
+            canvases.files?.map((file) => ({
+              channelId: lastSelectedChannel.id,
+              url: file.url_private,
+              title: file.title,
+              originId: file.id,
+              createdAt: file.created,
+              updatedAt: file.updated,
+              dataType: "canvas",
+            })) ?? [],
+        };
+
+        setSelectedChannels((prev) => [...prev.slice(0, -1), updatedChannel]);
+      }
+    }
+  }, [canvases]);
+
+  const handleChannelSelect = (channels: SlackChannel[]) => {
+    setSelectedChannels(channels);
+  };
+
+  const isLoading = isChannelsLoading;
```

Comment:
should this not be OR of every other loading variable? Smth like `isChannelLoading || isMessagesLoading ...` etc.